public/components/trace_analytics/components/common/plots/service_map.tsx

```diff
@@ -124,9 +132,21 @@ export function ServiceMap({
     },
   ];
 
+  useEffect(() => {
+    if (items?.graph?.nodes) {
+      const visibleNodes = items.graph.nodes.map((node) => node.label);
+      const options = Object.keys(serviceMap)
+        .filter((key) => visibleNodes.includes(serviceMap[key].serviceName))
+        .map((key) => ({
+          label: serviceMap[key].serviceName,
+          value: serviceMap[key].serviceName,
+        }));
+      setSelectableOptions(options);
+    }
+  }, [items.graph, serviceMap]);
```

Comment:
if `items` is undefined or doesn't have the `graph` field inside it, wouldn't this throw an error that would crash the page? i see that `items` is initialized to an empty object, so it seems like this might be a possibility.